### ISSUE TITLE
Cannot suppress OvershadowingSubclassFields on individual fields

### ISSUE THREAD

**Issue by msridhar** [google/error-prone#668]
Example:
```java
class Foo {
    static class A { int f; }

    static class B extends A { @SuppressWarnings("OvershadowingSubclassFields") int f; }
}
```

Even with the suppression annotation, a warning is still reported.  The suppression needs to be moved to the enclosing `B` class in order to actually suppress the warning.

This is happening because the errors are being reported from a [`matchClass` method](https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/bugpatterns/OvershadowingSubclassFields.java#L62).  I think the "right" fix would be to implement `VariableTreeMatcher` and report the errors from there (maybe computing the info for the enclosing class once and then caching somehow).  At the least, the documentation should be clarified.

**Comment by PhilippWendler** [comment]
Additionally, I would like to suggest to accept `@SuppressWarnings("hiding")` as suppression method for this check, too, because this is what Eclipse uses.

**Comment by eaftan** [comment]
@PhilippWendler "hiding" was added as an alternate suppression string in 2ecd9d09e17ae5bf2e22da882cffd2c2acbe07eb.
